src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,107 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static int DEFAULT_CAPACITY = 10;
+    private Object[] values;
+    private int size;
+
+    public ArrayList() {
+        values = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
+        if (size == values.length) {
+            resize();
+        }
+        values[size] = value;
+        size++;
 
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndex(index);
+        if (size == values.length) {
+            resize();
+        }
+        System.arraycopy(values, index, values, index + 1,size - index);
+        values[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index + 1);
+        return (T) values[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index + 1);
+        values[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index + 1);
+        T elementToRemove = (T) values[index];
+        size--;
+        System.arraycopy(values,index + 1, values, index,size - index);
+        return elementToRemove;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i <= size; i++) {
```

Comment:
```suggestion
        for (int i = 0; i < size; i++) {
```
why did you change it in previous commit?
it was fine

Reply:
Because in case i < size, exception was thrown in method get(int index) even if index was 100% right. So <= helped to solve the problem.